src/main/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/pullbased/ResultBlockIterator.java

```diff
@@ -7,4 +7,5 @@
 public interface ResultBlockIterator extends Iterator<IntermediateResultBlock>
 {
 
+    ResultElementIterator getElementIterator();
```

Comment:
Such a method should not be part of this interface. Not all types of ResultBlockIterator will be based on a ResultElementIterator. For instance, there will be a type of ResultBlockIterator that is based on a temporary index that is built on the fly for the solution mappings produced/retrieved by some operator.